spec/controllers/api/v1/book_suggestions_controller_spec.rb

```diff
@@ -0,0 +1,199 @@
+          price: price,
+          title: title,
+          link: link,
+          publisher: publisher,
+          year: year,
+          user_id: user.id
+        }
+      end
+
+      subject(:create_request) { post :create, params: params }
+
+      it 'book suggestions not change count' do
+        expect { create_request }.not_to(change { BookSuggestion.count })
+      end
+
+      it 'responds with bad_request status' do
+        create_request
+        expect(response).to have_http_status(:bad_request)
+      end
+
+      it 'returns error messages' do
+        create_request
+        expect(response.body['error']).to be_present
+      end
+    end
+
+    context 'When create a book suggestion without title' do
+      let!(:editorial) { Faker::Book.publisher }
+      let!(:price)     { Faker::Commerce.price }
+      let!(:author)    { Faker::Book.author }
+      let!(:link)      { Faker::Internet.url }
+      let!(:publisher) { Faker::Book.publisher }
+      let!(:year)      { Faker::Number.between(1950, Time.zone.today.year) }
+      let!(:user)      { create(:user) }
+
+      let!(:params) do
+        {
+          editorial: editorial,
+          price: price,
+          author: author,
+          link: link,
+          publisher: publisher,
+          year: year,
+          user_id: user.id
+        }
+      end
+
+      subject(:create_request) { post :create, params: params }
+
+      it 'book suggestions not change count' do
+        expect { create_request }.not_to(change { BookSuggestion.count })
+      end
+
+      it 'responds with bad_request status' do
+        create_request
+        expect(response).to have_http_status(:bad_request)
+      end
+
+      it 'returns error messages' do
+        create_request
+        expect(response.body['error']).to be_present
+      end
+    end
+
+    context 'When create a book suggestion without link' do
+      let!(:editorial) { Faker::Book.publisher }
+      let!(:price)     { Faker::Commerce.price }
+      let!(:author)    { Faker::Book.author }
+      let!(:title)     { Faker::Book.title }
+      let!(:publisher) { Faker::Book.publisher }
+      let!(:year)      { Faker::Number.between(1950, Time.zone.today.year) }
+      let!(:user)      { create(:user) }
+
+      let!(:params) do
+        {
+          editorial: editorial,
+          price: price,
+          author: author,
+          title: title,
+          publisher: publisher,
+          year: year,
+          user_id: user.id
+        }
+      end
+
+      subject(:create_request) { post :create, params: params }
+
+      it 'book suggestions not change count' do
+        expect { create_request }.not_to(change { BookSuggestion.count })
+      end
+
+      it 'responds with bad_request status' do
+        create_request
+        expect(response).to have_http_status(:bad_request)
+      end
+
+      it 'returns error messages' do
+        create_request
+        expect(response.body['error']).to be_present
```

Comment:
🤔 this is getting repetitive and long, right?
Why not use a general context for the creation where you set everything up and then just add little contexts inside that, only modifying the `params` variable.

Reply:
:thinking: I will try :smile: